src/main/java/org/alfresco/heartbeat/InfoDataCollector.java

```diff
@@ -129,6 +178,54 @@ public void afterPropertiesSet() throws Exception
         infoValues.put("edition", serverDescriptor.getEdition());
         infoValues.put("deploymentMethod", deploymentMethodProvider.getDeploymentMethod().toString());
 
+        infoValues.put("os.vendor", System.getProperty("os.name"));
+        infoValues.put("os.version", System.getProperty("os.version"));
+        infoValues.put("os.arch", System.getProperty("os.arch"));
+        infoValues.put("java.vendor", System.getProperty("java.vendor"));
+        infoValues.put("java.version", System.getProperty("java.version"));
+
+        infoValues.put("user.language", Locale.getDefault().getLanguage());
+        infoValues.put("user.timezone", displayTimeZone(TimeZone.getDefault()));
+
+        try
+        {
+            infoValues.put("server.info", servletContext.getServerInfo()); 
+        }
+        catch (NullPointerException e)
+        {
+            infoValues.put("server.info", null);
+        }
+        
+        Connection con = null;
+        try
```

Comment:
Use `try-with-resources` please, makes the code more compact and easier to follow.

Reply:
Changes in the next commit.